android/src/main/java/com/it_nomads/flutter_realm/FlutterRealm.java

```diff
@@ -31,11 +32,25 @@
 
         RealmConfiguration.Builder builder = new RealmConfiguration.Builder().modules(Realm.getDefaultModule());
 
+        byte[] encryptionKey = (byte[]) arguments.get("encryptionKey");
+        String fileDirectory = (String) arguments.get("fileDirectory");
+        String fileName = (String) arguments.get("fileName");
         String inMemoryIdentifier = (String) arguments.get("inMemoryIdentifier");
 
-        if (inMemoryIdentifier == null) {
-        } else {
+        if (inMemoryIdentifier != null) {
             builder.inMemory().name(inMemoryIdentifier);
+        } else if (fileDirectory != null || fileName != null) {
+            if (fileDirectory != null) {
+                RealmConfiguration tmp = builder.build();
+                File file = new File(tmp.getRealmDirectory(), fileDirectory);
```

Comment:
I'm not sure if `tmp.getRealmDirectory()` is the most efficient way to get the default Realm directory. I would prefer to call `builder.build()` only once...

Reply:
What is the case for custom directory?

Reply:
From Realm:

> The most common location to store writable Realm files is the “Documents” directory on iOS and the “Application Support” directory on macOS. Please respect Apple’s iOS Data Storage Guidelines, which recommend that if documents that can be regenerated by the app should be stored in the <Application_Home>/Library/Caches directory. If a custom URL is used to initialize a Realm, it must describe a location with write permissions.

> For example, suppose you have an application where users have to log in to your web backend, and you want to support quickly switching between accounts. You could give each account its own Realm file that will be used as the default Realm by doing the following...

I think two use cases for custom directories:

1. Your app needs to cache some non-user generated data and the cached data can be regenerated. The realm should technically be in the `.../Library/Caches` directory on iOS. Note that this PR does not support paths other than the standard documents directory on iOS or Android. If someone needs that flexibility in the future I'd suggest just adding a `cacheDirectory` property uses the appropriate base directory on iOS and Android.
1. My app allows multiple users on a single device which each user belonging to 1 or more projects. Data is sensitive and each user/project must have its own realm.
```
/user1/projectA/project.realm
/user1/projectA/additional_project_files...
/user1/projectB/project.realm
/user1/projectB/additional_project_files...
/user2/projectA/project.realm
etc...
```

Reply:
How about defining full path (including file name) in Dart code? 

It simplifies plugin (native, duplicated) code in iOS and Android and resolves problems what if I want to use TMP, DOC,CACHE or another folder.

Reply:
I can do that. I started out that way but then switched to the current approach. My thought was that 95% of the time a developer would want to write to the DOC directory. Current approach made it so you don't need to use [path_provider](https://pub.dev/packages/path_provider). Full path is more flexible though.

Side note, it may be awhile before I get around to making changes.